src/pykeen/pipeline.py

```diff
@@ -666,7 +666,7 @@ def pipeline(  # noqa: C901
     if model_kwargs is None:
         model_kwargs = {}
     model_kwargs.update(preferred_device=device)
-    model_kwargs.setdefault('random_seed', NoRandomSeedNecessary)
+    model_kwargs.setdefault('random_seed', random_seed)
```

Comment:
@mberr @mali-git I was definitely the one who implemented this part of the pipeline but for the life of me I can't remember if there was a reason for NoRandomSeedNecessary other than making the HPO not emit a warning message... if not then let me know if you think there are reprecussions for setting this in the config *post facto*

Reply:
If I remember correctly, the reason was that we already set the random seed in:
https://github.com/pykeen/pykeen/blob/22f8815fe6520dc89f786e43d5cb3682abe1113d/src/pykeen/pipeline.py#L642

If we pass `random_seed` to the model, we set it again in:
https://github.com/pykeen/pykeen/blob/22f8815fe6520dc89f786e43d5cb3682abe1113d/src/pykeen/models/base.py#L247

which will cause difficulties in reproducibility.

Reply:
correct, but that only happens if there is no random seed already set so there's both the possibility to do this through the pipeline and so you could roll your own pipeline